plugins/inputs/tencentcloudcm/tencentcloudcm.go

```diff
@@ -0,0 +1,448 @@
+	# Note that if a period is configured that is smaller than the default for a
+	# particular metric, that metric will not be returned by the Tencent Cloud API
+	# and will not be collected by Telegraf.
+	#
+	# Requested Tencent Cloud Cloud Monitor aggregation Period (required - must be a multiple of 60s)
+	period = "1m"
+  
+	# Collection Delay (must account for metrics availability via Tencent Cloud API)
+	delay = "5m"
+  
+	## Maximum requests per second. Note that the global default Tencent Cloud API rate limit is
+	## 20 calls/second (1,200 calls/minute), so if you define multiple namespaces, these should add up to a
+	## maximum of 20.
+	## See https://intl.cloud.tencent.com/document/product/248/33881
+	ratelimit = 1000
+  
+	# Timeout for http requests made by the Tencent Cloud client.
+	timeout = "5s"
+  
+	## By default, Tencent Cloud CM Input plugin will automatically discover instances in specified regions
+	## This sets the interval for discover and update the instances discovered.
+	##
+	## how often the discovery API call executed (default 1m)
+	# discovery_interval = "1m"
+  
+	# Tencent Cloud Account (required - you can provide multiple entries and distinguish them using
+	# optional name field, if name is empty, index number will be used as default)
+	[[inputs.tencentcloudcm.accounts]]
+	  name = ""
+	  secret_id = ""
+	  secret_key = ""
+  
+	  # Namespaces to Pull
+	  [[inputs.tencentcloudcm.accounts.namespaces]]
+		# Tencent Cloud CM Namespace (required - see https://intl.cloud.tencent.com/document/product/248/34716#namespace)
+		namespace = "QCE/CVM"
+  
+		# Metrics filter, all metrics will be pulled if not left empty. Different namespaces may have different
+		# metric names, e.g. CVM Monitoring Metrics: https://intl.cloud.tencent.com/document/product/248/6843
+		# metrics = ["CPUUsage", "MemUsage"]
+  
+		[[inputs.tencentcloudcm.accounts.namespaces.regions]]
+		  # Tencent Cloud regions (required - Allowed values: https://intl.cloud.tencent.com/document/api/248/33876)
+		  region = "ap-guangzhou"
+  
+		  ## Dimension filters for Metric. Different namespaces may have different
+		  ## dimension requirements, e.g. CVM Monitoring Metrics: https://intl.cloud.tencent.com/document/product/248/6843It must be specified if the namespace does not support instance auto discovery
+		  ## Currently, discovery supported for the following namespaces:
+		  ## - QCE/CVM
+		  ## - QCE/CDB
+		  ## - QCE/CES
+		  ## - QCE/REDIS
+		  ## - QCE/LB_PUBLIC
+		  ## - QCE/LB_PRIVATE
+		  # [[inputs.tencentcloudcm.accounts.namespaces.regions.instances]]
+		  # [[inputs.tencentcloudcm.accounts.namespaces.regions.instances.dimensions]]
+		  #   name = ""
+		  #   value = ""
+`
+}
+
+// Description implements telegraf.Input interface
+func (t *TencentCloudCM) Description() string {
+	return "Pull Metric Statistics from Tencent Cloud Cloud Monitor"
+}
+
+// Init is for setup, and validating config.
+func (t *TencentCloudCM) Init() error {
+
```

Comment:
No whitespaces at the beginning of the function please. Use whitespaces to structure the code i.e. to improve readability and help the reading flow.

Reply:
fixed all redundant whitespaces.